django/db/models/query.py

```diff
@@ -485,8 +485,7 @@ def update_or_create(self, defaults=None, **kwargs):
         for k, v in six.iteritems(defaults):
             setattr(obj, k, v)
 
-        with transaction.atomic(using=self.db, savepoint=False):
-            obj.save(using=self.db)
+        obj.save(using=self.db)
```

Comment:
Indeed this one looks fine.

It will have beneficial performance effects if:

- ATOMIC_REQUESTS is not in use
- `update_or_create` ends up creating an object
-  the `save` method of that objects does long things like sending email *after* the write (i.e. once it holds an exclusive lock on the database)